### ISSUE TITLE
🐛[BUG]: AIFS2 and AIFS2ENS assign time-dependent forcings to the wrong samples for multiple batches and initialization times

### ISSUE THREAD

**Issue by gertln** [NVIDIA/earth2studio#1008]
### Version

0.17.0a0 (main)

### On which installation method(s) does this occur?

source

### Describe the issue

`AIFS2` and `AIFS2ENS` place their time-dependent forcings (solar insolation, Julian day, local time) on the wrong samples when run with more than one batch/ensemble member **and** more than one initialization time. The `(batch, time)` axis is flattened batch-major (flattened slot `= b * n_time + t`), but the forcing loop writes each time's forcing into a contiguous time-major block, so off-diagonal `(batch, time)` samples receive another sample's forcings.

## Location

- `earth2studio/models/px/aifs2ens.py` — `_prepare_input` and `_update_input`
- `earth2studio/models/px/aifs2.py` — `_prepare_input` and `_update_input`

In each, the loop over times writes to `x[t * n_batch : (t + 1) * n_batch]` (time-major), while the tensor is flattened batch-major.

## Reproduction

The defect is pure index bookkeeping, so it reproduces with only `torch` (no model weights). The model flattens `[n_batch, n_time, ...]` batch-major (slot `= b * n_time + t`), but the forcing loop writes time `t` into the contiguous time-major block `[t * n_batch : (t + 1) * n_batch]`:

```python
import torch

n_batch, n_time = 2, 2  # >1 ensemble member AND >1 init time

# Forcing loop as written: time t -> contiguous block (time-major)
assigned = torch.empty(n_batch * n_time, dtype=torch.long)
for t in range(n_time):
    assigned[t * n_batch : (t + 1) * n_batch] = t

# But the flattened slot is batch-major: slot = b * n_time + t
for bt in range(n_batch * n_time):
    b, t = bt // n_time, bt % n_time
    ok = assigned[bt].item() == t
    print(f"slot {bt} = (b{b}, t{t}): got forcing for time{assigned[bt].item()}  "
          f"{'OK' if ok else 'WRONG'}")
```

Output: 2/4 samples receive the wrong initialization time's forcings:

```
slot 0 = (b0, t0): got forcing for time0  OK
slot 1 = (b0, t1): got forcing for time0  WRONG
slot 2 = (b1, t0): got forcing for time1  WRONG
slot 3 = (b1, t1): got forcing for time1  OK
```

Confirmed against the real `_prepare_input`/`_update_input` on a small grid at 00Z vs 12Z, the mis-assigned samples receive the wrong time's solar-zenith field (day/night swapped). The exact affected pattern depends on `n_batch` and `n_time`.

## Impact

For `AIFS2` or `AIFS2ENS` runs with more than one batch/ensemble member and more than one initialization time, the corrupted forcings feed the model and silently produce wrong forecasts for the mis-assigned samples. Runs with a single initialization time or a single member are unaffected (the two orderings coincide).

## Expected behavior

Each `(batch, time)` sample should receive the forcings computed for its own initialization time.
